File: leetcode/easy/MaximumSubarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max = nums[0];
        int sum = nums[0];
        for (int i = 1; i < nums.length; i++) {
            int acc = sum + nums[i];
            sum = acc > nums[i] ? acc : nums[i];
            if (max < sum) {
                max = sum;
            }
        }
        return max;
    }
}

File: leetcode/easy/MaximumSubarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        m = nums[0];
        s = nums[0];
        for i in range(1, len(nums)):
            s = max(s + nums[i], nums[i])
            if m < s:
                m = s
        return m
